ethcore/src/snapshot/account.rs

```diff
@@ -105,9 +95,47 @@ pub fn to_fat_rlps(acc: &BasicAccount, acct_db: &AccountDB, used_code: &mut Hash
 			}
 		}
 
-		account_stream.append_raw(&pairs_rlp, 1);
-		Ok(account_stream.out())
-	}).collect()
+		account_stream.begin_unbounded_list();
+		if account_stream.len() > target_chunk_size {
+			// account does not fit, push an empty record to mark a new chunk
+			target_chunk_size = max_chunk_size;
+			chunks.push(Vec::new());
+		}
+
+		if let Some(pair) = leftover.take() {
+			account_stream.append_raw_checked(&pair, 1, target_chunk_size);
+		}
+
+		loop {
+			match db_iter.next() {
```

Comment:
one of those loops could be moved to a new structure -> `RlpChunkedStream` which returns vec of chunks